atlp-rwanda/Phantom-Backend-Pirates
# Ft assign driver bus

## PR Introduction
* As an operator, I want to assign a driver to a bus, so that they can control it.

## Description of the task
* The operator should be able to assign a driver to a bus
* The operator should be able to view a paginated list of the driver to bus assignment
* The driver should receive  a bus assignment and unassignment email & in-app notifications
* Appropriate messages are displayed on unsuccessful attempts

## How should me manually tested
* Clone this branch and cd into it
* Install all the packages used by running `npm install`
* Migrate tables by running in your terminal `npm run migrate`
*  Run seeds using `npm run seed ` command
* Run in your terminal `npm intsall nodemailer`  a package which will be used to receive notifications
* Using postman test this login api using this endpoint `users/login`
* Set a token received into header for authorization 
* Using postman test this assigning bus to driver api using this endpoint `api/buses/:busId/:employeeId`
* If the employee is  a driver should and the bus has no driver assigned to it the previous test should succeed and a driver receive a notification email on nodemailer
* Using postman view a paginated list of the driver to bus assignment using this endpoint `api/drivers/buses`
* Using postman unassign a driver from bus usind this endpoint `/api/buses/unassign/:busId`
* A driver also should receive a unassignment email
